src/stlink-lib/chipid.c

```diff
@@ -55,29 +55,23 @@ void process_chipfile(char *fname) {
 
     ts = calloc(sizeof(struct stlink_chipid_params), 1);
 
-    while (fgets(buf, 1024, fp) != NULL) {
-        for (p = buf; isspace (*p); p++);
+    while (fgets(buf, sizeof(buf), fp) != NULL) {
 
-        if (!*p) {
-            continue; // we hit end-of-line with only whitespace
-        }
```

Comment:
If there’s an empty line between two valid lines, wouldn’t we process the first one twice?

Reply:
Just to clarify: You mean something like this?
```
# Chip-ID file for STM32F401xD/xE device
#
dev_type STM32F401xD_xE

ref_manual_id 0368
chip_id 0x433                // STM32_CHIPID_F4_DE
flash_type F2_F4
...
```

Reply:
If that is the case you meant it processes the file properly after testing that case.

Reply:
Well, it seems it processes the properly, but it seems I was right that every blank line causes it to process the last non-blank line again. I think we it would be better to avoid such behavior.

Reply:
What about an additional check if the line is empty? 3b328b26bd5eb0e07f78b8ba48e10c2581a1239a

Reply:
Yes, empty lines should be detected and simply ignored from my point of view.

Reply:
This point appears to remain unresolved yet...

Reply:
Do you mean even after 3b328b2? Can you point into the direction where i missed the point?

Reply:
No, that should be fine, as long as it finds it's way into this PR.